client/state/current-user/reducer.js

```diff
@@ -82,9 +90,28 @@ export const capabilities = createReducer( {}, {
 	}
 }, capabilitiesSchema );
 
+export const gravatarStatus = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_START ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: true
+		} );
+	},
+	[ GRAVATAR_UPLOAD_SUCCESS ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+	[ GRAVATAR_UPLOAD_FAILURE ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+}, gravatarSchema );
```

Comment:
The images might get large, so I'll size them down to the maximum size we use in Calypso (400px right now). I'll also dispatch an action to remove the image from the state whenever the date has passed (from inside a check in `<Gravatar />`).

We can get into trouble with this later if we don't display the current user's Gravatar for a long time, but it's pretty ubiquitous right now and will continue to be, so I think the approach is sensible.

> I think this is maybe work adding in our own subtree.

I agree. I'll need to do that in order to persist the new image information, but not the loading state.